sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -671,6 +673,8 @@ def get_media_settings_value(physical_port, key):
 
         if key[0] in media_dict:
             return media_dict[key[0]]
+        elif key[0].split('-')[0] in media_dict:
+            return media_dict[key[0].split('-')[0]]
```

Comment:
@amnsinghal i think line 678 check if PN is matching which is not as per the HLD?

Reply:
key[] is a list of [vendor_key, media_key], so key[0] = vendor_key = "<vendor_name>-<vendor_PN>".
This change still checks for <vendor_name>-<vendor_PN> first (line 674), then "<vendor_name>" alone (line 676), then <media_key> and lastly 'default' key.